packages/jellyfish-api-core/__tests__/category/icxorderbook/common.test.ts

```diff
@@ -0,0 +1,139 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import {
+  ICXOrderInfo, ICXOrder
+} from '../../../src/category/icxorderbook'
+import { createToken, mintTokens, accountToAccount } from '@defichain/testing'
+import { BigNumber } from '@defichain/jellyfish-api-core'
+
+// globals
+export let symbolDFI: string
+export let symbolBTC: string
+export let accountDFI: string
+export let accountBTC: string
+export let poolOwner: string
+export let idDFI: string
+export let idBTC: string
+export let ICX_TAKERFEE_PER_BTC: number
+export let DEX_DFI_PER_BTC_RATE: number
+
+export async function setup (container: MasterNodeRegTestContainer): Promise<void> {
```

Comment:
- you should write `aspect oriented` setup if you want to maintain a list of common setups. (`_setups.ts`)
- you should separate `expects` out into `_expects.ts`

Reply:
need more clarifications on this. are you saying that all setups and common functions should be in `_setups.ts` file?
what about the second point?

Reply:
The first point is to have your setups in `icx_setup.ts` and keep them functional aspect-oriented setup.

```ts
class ICXSetup {
  constructor(private MasternodeRegTestContainer container){}
  
  createToken(...) {
    this.container....
  }
  
  mintToken(symbol: string, amount: number) {
    ...
  }
}
```

Within your `beforeAll() {}`:

1. `setup.createToken()`
2. `setup.mintToken()`
3. ... all other setups

Because it's aspect-based, this makes it much easier to read what is your setup for that describes context without cross-referencing the commons file.

Reply:
The second point is since you want to share assertions, you should use the `jest.io` native test `expect` template.

> https://jestjs.io/docs/expect

Check out the code below, you can declare a matcher template that can be reused for all tests. (meaning instead of doing assertions for each field you can provide a template and re-use for all tests)

https://github.com/DeFiCh/whale/blob/ced01e11d766d72b45dabad05f712a990d3e1e29/src/module.api/address.controller.e2e.ts#L109-L131